ports/mimxrt/main.c

```diff
@@ -222,6 +223,9 @@ int main(void) {
 
 soft_reset_exit:
     mp_printf(MP_PYTHON_PRINTER, "MPY: soft reboot\n");
+    #if OMV_ENABLE_VOSPI || OMV_LEPTON_ENABLE
+    vospi_deinit();
+    #endif
```

Comment:
I was going to add a finalizer to CSI-NG. This is how make_new works now:
```C
    omv_csi_t *csi = omv_csi_get(args[ARG_id].u_int);
....
    if (csi->fb == NULL) {
      /// allocate FB
```
But `csi->fb` will be a dangling pointer after a soft-reset, so the finalizer will just set it to NULL so it gets allocated again. I could also `omv_csi_abort` in the finalizer, which in turn calls `csi->abort` which we could implement for lepton to deinit vospi. What do you think?

Reply:
Yes, omv_csi_abort() is the way to go.